### PR TITLE
bugfix/13749-negative-ticks-on-positive-axes

### LINKED ISSUE #13749: Issue with negative ticks caused by using tickAmount and min together
#### Expected behaviour
When no negative values exist in data set, the expectation is that the x and y axes must start at 0 (min: 0 is given) and must plot only the positive ticks (tickAmount: 5 is given). Such as shown below using Highcharts version 5:
Correct-version5

#### Actual behaviour
Highcharts upgrade from version 5 to version 7 is causing negative ticks on x and y axes even when no negative data exists. Various chart plots are wrongly displaying negative axes. This worked perfectly with version 5 but is behaving weird with version 7 as shown below:
Issue

Look at the starting tick, -2.5, whereas there are no negative values, so it must start at 0. Also observe the last tick is not appropriate width as the other tick intervals and ends abruptly in between.

#### Live demo with steps to reproduce
Here is the link for demonstration of the issue: https://jsfiddle.net/3awvyn0b/3/, see config for x-axis which is not being reflected in the plot:
xAxis: {
        tickAmount: 5,
        min: 0
    },
If I remove tickAmount for the x-axis, the negative ticks are resolved but then the ticks are not correctly marked. See here:
Capture1

#### Product version
Highcharts 7

#### Affected browser(s)
Chrome

### PR DESCRIPTION
**Pull request by mateuszkornecki**
Fixed #13749, negative ticks on positive axes with the `xAxis.min` = 0.

### CODE REVIEW

File: ts/Core/Axis/Axis.ts [hunk 1]
```diff
@@ -5899,27 +5911,29 @@ class Axis implements AxisComposition, AxisLike {
...
+            threshold = pick(axis.threshold, axis.softThreshold ? 0 : null);
+
+        let min, len, i;
 
         if (axis.hasData()) {
             if (currentTickAmount < tickAmount) {
                 min = axis.min;
 
                 while (tickPositions.length < tickAmount) {
+                    const isMinHigherThanTickInterval = (min || min === 0) && (min < tickInterval);
```
Comment (pawelfus) [inline ts/Core/Axis/Axis.ts:5931]: Probably should be `Lower`, not `Higher`

Comment (mateuszkornecki) [reply ts/Core/Axis/Axis.ts:5931]: Indeed setting the `x Axis.softThreshold` will resolve this issue, unfortunately, it is causing more than 100 visual changes, and breaking a few tests.

**Comment by mateuszkornecki** [comment]
> Just an idea: It works fine for yAxis, does not work for xAxis. It seems that yAxis gets its threshold (or softThreshold) from series. xAxis by default has both props disabled (or nullified), but if we set them, everything seems to be fine: https://jsfiddle.net/BlackLabel/3pu6axed/2/
> 
> Can we explore this solution? What do you think @KacperMadej @mateuszkornecki ?

**Comment by mateuszkornecki** [comment]
> Just an idea: It works fine for yAxis, does not work for xAxis. It seems that yAxis gets its threshold (or softThreshold) from series. xAxis by default has both props disabled (or nullified), but if we set them, everything seems to be fine: https://jsfiddle.net/BlackLabel/3pu6axed/2/
> 
> Can we explore this solution? What do you think @KacperMadej @mateuszkornecki ?

Indeed enabling the `softThreshold` for `xAxis` will resolve this issue, unfortunately, it is causing more than 100 visual changes, and breaking a few tests.

**Comment by mateuszkornecki** [comment]
One visual test is currently failing. I discussed that edge-case with @pawelfus and it seems that both results are correct. I can prevent changing axes with parallel coordinates by adding a simple condition, but I am not sure if it's really necessary.

**Comment by TorsteinHonsi** [comment]
I think we need to reconsider how to fix this - the original fix for #3965 also seems to work only by accident. If the data is all negative, the `softThreshold` is not respected. Here's what should happen:
* If `min` is set, we should not prepend ticks below `min`. Likewise for `max`.
* If `softThreshold`, and the data is all-positive, we should not prepend ticks below `min`. Equivalent for all-negative data.

I'm creating a new PR with this approach.
